src/pages/index.js

```diff
@@ -4,73 +4,201 @@ import Section from "../components/Section.js";
-  userInformation.setUserInfo({
-    title: data.header,
-    description: data.description,
+  const userInformation = new UserInfo({
+    name: ".profile__header",
+    description: ".profile__description",
+    avatar: ".profile__avatar",
   });
-});
-profileEditForm.setEventListeners();
-
-constants.profileEditButton.addEventListener("click", () => {
-  profileEditValidator.resetValidation();
-  const userData = userInformation.getUserInfo();
-  constants.profileTitleInput.value = userData.title;
-  constants.profileDescriptionInput.value = userData.description.trim();
-  profileEditForm.open();
-});
 
-const addCardForm = new PopupWithForm("#add-card-popup", (data) => {
-  cardSection.addItem(createCard({ name: data.title, link: data.URL }));
-  addCardValidator.disableButton();
-});
-addCardForm.setEventListeners();
+  let cardSection;
+  api
+    .getAppInfo()
+    .then(([userInfo, initialCards]) => {
+      userInformation.setUserInfo(userInfo);
+      userInformation.setUserAvatar(userInfo);
+      if (Array.isArray(initialCards)) {
+        cardSection = new Section({ renderer: renderCard }, ".cards__list");
+        cardSection.renderItems(initialCards);
+      } else {
+        console.error("Initial cards data is not an array:", initialCards);
+      }
+    })
+    .catch((err) => {
+      console.error(err);
+    });
 
-constants.addCardButton.addEventListener("click", () => {
-  addCardForm.open();
-});
+  function renderCard(cardData) {
+    const card = new Card(
+      cardData,
+      "#card-template",
+      handleImageClick,
+      handleCardDeleteClick,
+      handleLikeClick
+    ).getView();
+    cardSection.addItem(card);
+  }
 
-const cardSection = new Section(
-  {
-    renderer: (item) => {
-      const cardElement = createCard(item);
-      cardSection.addItem(cardElement);
-    },
-  },
-  constants.cardListEl
-);
+  const cardPreview = new PopupWithImage("#image-popup");
+  const confirmDeletePopup = new PopupWithConfirm("#delete-card-popup");
+  cardPreview.setEventListeners();
+
+  const profileEditValidator = new FormValidator(
+    constants.config,
+    document.querySelector("#profile-form")
+  );
+  const addCardValidator = new FormValidator(
+    constants.config,
+    document.querySelector("#add-card-form")
+  );
+  const avatarValidator = new FormValidator(
+    constants.config,
+    document.querySelector("#avatar-form")
+  );
+
+  profileEditValidator.enableValidation();
+  addCardValidator.enableValidation();
+  avatarValidator.enableValidation();
+
+  const avatarImagePopup = new PopupWithForm("#edit-avatar-popup", (data) => {
+    avatarImagePopup.renderLoading(true);
+
+    api
+      .updateAvatar(data.link)
+      .then((res) => {
+        userInformation.setUserAvatar(res);
+      })
+      .catch((err) => {
+        console.error("Error updating avatar:", err);
+      })
+      .finally(() => {
+        avatarImagePopup.renderLoading(false);
+        avatarImagePopup.close();
```

Comment:
```js

        avatarImagePopup.close();
        avatarImagePopup.resetForm();
        avatarValidator.toggleButtonState();
```

You shouldn't have placed closing and resetting in `finally`. You need to add `toggleButtonState` only on **opening** to toggle the button 


```js

    api
      .updateAvatar(data.link)
      .then((res) => {
        userInformation.setUserAvatar(res);
        avatarImagePopup.close();  // close
        avatarImagePopup.resetForm();  // reset
      })
      .catch((err) => {
        console.error("Error updating avatar:", err);
      })
      .finally(() => {
        avatarImagePopup.renderLoading(false);  // finally
      });
  });

```


**Please, correct it everywhere**